### ISSUE TITLE
CSI volume source fstype incompatible with CSI spec

### ISSUE THREAD

**Issue by msau42** [kubernetes/kubernetes#65122]
**Is this a BUG REPORT or FEATURE REQUEST?**:
@kubernetes/sig-storage-bugs 

**What happened**:
CSI spec allows for an empty fstype string.  This is intended for non-block plugins like nfs and gluster where filesystems are not separately created on the volume.
```
  // Indicate that the volume will be accessed via the filesystem API.
  message MountVolume {
    // The filesystem type. This field is OPTIONAL.
    // An empty string is equal to an unspecified field value.
    string fs_type = 1;
  }
```

The Kubernetes CSI volume source API defaults fstype to "ext4" and doesn't allow an empty string:
```
        // Filesystem type to mount.
        // Must be a filesystem type supported by the host operating system.
        // Ex. "ext4", "xfs", "ntfs". Implicitly inferred to be "ext4" if unspecified.
        // +optional
        FSType string
```

This makes it impossible for Kubernetes to pass in an empty fs_type to the plugin.  

**What you expected to happen**:
Since we currently leave fs creation and formatting up to the plugin, the plugin should be the one to decide the default.

**Comment by saad-ali** [comment]
We should change the default `FSType` value for `CSIPersistentVolumeSource` empty string. And plugins will be responsible for setting a sane default.

**Comment by msau42** [comment]
From the CSI side though, it is unclear from the spec what the plugin should do if fsType is empty and it requires one

**Comment by saad-ali** [comment]
> From the CSI side though, it is unclear from the spec what the plugin should do if fsType is empty and it requires one

The plugin can decide what makes sense for it, since the spec does not dictate.

**Comment by msau42** [comment]
Fixed by #65499
/close
